client/src/DiscInfo/DiscInfo.js

```diff
@@ -1,71 +1,50 @@
 import React, {Component} from 'react';
 import PropTypes from 'prop-types';
-import $ from 'jquery';
 
-import {
-    Button,
-    Form,
-    FormGroup,
-    Input,
-    Label,
-    Table,
-} from 'reactstrap';
-
-import {
-    actionRipTracks,
-} from '../api.js'
+import {Button, Form, FormGroup, Input, Label, Table,} from 'reactstrap';
 
+import {actionRipTracks,} from '../api.js'
 
 class DiscInfo extends Component {
-
     constructor(props) {
         super(props);
-        let selectedTracks = {};
-        this.props.tracks && this.props.tracks.map((trackInfo, trackId) => {
-            selectedTracks[trackId] = trackInfo.isAutoSelected;
-        });
         this.state = {
             checkAll: false,
             discName: false,
-            selectedTracks: selectedTracks,
+            selectedTracks: this.props.tracks.map((trackInfo) => {
+                return trackInfo;
+            })
         };
     }
 
-    getTrackCheckboxes($formElement) {
-        return $formElement
-            .closest('fieldset')
-            .find('input[name=selectTrack]');
-    }
-
     // Toggle the checkbox on all tracks.
     toggleAllTracks(event) {
-        let $target = $(event.target);
-        this.setState({
-            checkAll: $target.prop('checked'),
-        });
-        this.getTrackCheckboxes(event.target)
-            .prop('checked', this.state.checkAll);
+        alert('This functionality is currently disabled.');
+        // let $target = $(event.target);
+        // this.setState({
+        //     checkAll: $target.prop('checked'),
+        // });
+        // this.getTrackCheckboxes(event.target)
+        //     .prop('checked', this.state.checkAll);
     }
 
-    toggleTrack(trackId) {
-        let changeObj = {};
-        changeObj[trackId] = !this.selectedTracks[trackId];
-        this.setState({
-            selectedTracks: Object.assign(
-                this.selectedTracks, changeObj
-            )
-        })
+    toggleTrack(event) {
+        let trackId = event.target.name;
+        this.state.selectedTracks[trackId].isSelected = event.target.checked;
     }
 
     // Command the server to rip certain tracks for this disc.
-    ripTracks(event) {
-        let ripTrackIds = this.getTrackCheckboxes(event.target)
-            .find(':checked')
-            .data('track-id');
+    ripTracks() {
+        let trackIds = [];
+        this.state.selectedTracks.forEach((selectedTrack, trackId) => {
+            if (selectedTrack.isSelected) {
+                trackIds.push(trackId);
+            }
+        });
```

Comment:
I was hoping to use a `.map(` function here, however for every track that I didn't return it would default to undefined. The result I wanted was something like: `['1', '5']`, however map would produce `[undefined, '1', undefined, undefined, undefined, '5']`. I'm wondering if a `filter` or `reduce` method would be a better option.

Reply:
Yep you want `reduce`